localstack/services/kms/models.py

```diff
@@ -554,7 +560,11 @@ def __init__(self, create_grant_request: CreateGrantRequest):
         # The Name field is present with just an empty string value.
         self.metadata.setdefault("Name", "")
 
-        self.token = long_uid()
+        # Encode account ID and region in grant token.
+        # This way the grant can be located when being retired by grant principal.
+        # The token consists of account ID, region name and a UUID concatenated with ':' and encoded with base64
+        decoded_token = account_id + ":" + region_name + ":" + long_uid()
+        self.token = to_str(base64.b64encode(to_bytes(decoded_token)))
```

Comment:
Similary, the account ID and region is now encoded in the grant token to help locate the grant when being operated on across-accounts.